Agents/MackayDataAgent/README.md

```diff
@@ -0,0 +1,129 @@
+```bash
+[rdb_access] # RDB Access
+db_url = 
+db_user = 
+db_pw = 
+
+[kg_access] # Triple Store Access
+endpoint = 
+
+[api_agent] # API Agent
+agent_url =
+agent_iri =
+
+[forecast_agent] # Forecast Agent
+agent_url = 
+agent_iri = 
+
+[output] # BASE URL of triples to create by this agent
+base_url = 
+```
+###1.1.2 Data Point Configuration
+Mackay Data Agents manage the update and reading of multiple data points, each the result of a chain of derivations. Configuration files one per data point are put under [data config folder]. New data points can easily be added by creating new files in the folder following the syntax below.
+```bash
+[output]
+steps = forecast # Derivation Step after API download, in [forecast|calculation]
+target_iri = https://test/Number_dwellingunit # IRI of target quantity
+api_iri = https://test/TSMap_dwellingunit # IRI of the API map instance
+name = dwelling_unit
+
+# Forecast parameters
+[forecast]
+predictEnd = 2050
+frequency = 1
+unitFrequency = year
+```
+
+
+## 1.2 Required Derivation Markup
+`Mackay Data Agent` is a client of the [API Agent]. If any api-meta-data IRI declared in the data config files is not registered to API agent and in turn never instantiated when the Mackay Data Agent starts, `Mackay Data Agent` will register them to the [API agent] (and subsequently the API data will be instantiated). This however is on the condition that the API meta-data triples are already properly instantiated in the KG.  Refer to [API Agent] README 1.3 on how to declare API meta-data in KG.
+
+`Mackay Data Agent` is also a client of the [Forecasting Agent]. However, there is no need for the user to manually insert the forecast meta-data triples, as the agent will automatically instantiate them according to the data config files.
+
+Note that this design choice is made as API-downloaded data are supposed to be used by multiple agents; therefore it is conceptually not sound to keep a copy of API meta-information in this one client agent.
+
+
+##1.3. Using the Agent
+
+The agent image can be built via:
+```
+docker compose -f "docker-compose-test-dockerised.yml" up -d --build
+```
+Note that the agent needs 4 other components to work:
+1. Blazegraph.
+2. PostgresSQL.
+3. [Forecasting Agent]
+4. [API Agent]
+#2. Agent Design
+
+# 2.1 Mackay Calculator Agents
+The Singapore MacKay Carbon Calculator provides a model of the Singapore energy system that allows you to explore pathways to decarbonisation. The scenario simulation is based on an underling Mackay Calculator Model. We implement it in the TWA agent framework with two agents.
+1) Mackay Calculator Agent: The agent that manages the web-interface and controlling API over the Calculator model.
+2) Mackay Data Agent: The agent that manages the update of the derived Timeseries (TS) datapoints in TWA that is required as the input to the Calculator Model. The required input TS data are derived under DIF with a chain of [Forecasting Agent] and [API Agent].
+
```

Comment:
So, can these two agents be run independently? Or is one dependent on the other? Please clarify. It would be helpful to state this explicitly.

Reply:
This is clarified in READMES of both agents.